pkg/resource/deploy/source_eval.go

```diff
@@ -596,6 +596,89 @@ func (rm *resmon) Invoke(ctx context.Context, req *pulumirpc.InvokeRequest) (*pu
 	return &pulumirpc.InvokeResponse{Return: mret, Failures: chkfails}, nil
 }
 
+// Call dynamically executes a method in the provider associated with a component resource.
+func (rm *resmon) Call(ctx context.Context, req *pulumirpc.CallRequest) (*pulumirpc.CallResponse, error) {
```

Comment:
This looks good. I'd love to see some inner-loop tests in `pkg/engine/lifecycletest`.